filebeat/crawler/prospector_log.go

```diff
@@ -251,14 +254,13 @@ func (p *ProspectorLog) checkExistingFile(h *harvester.Harvester, newFile *input
 		// We only need to keep it for the remainder of this iteration then we can assume it was deleted and forget about it
 		p.missingFiles[h.Path] = oldFile.FileInfo
 
-	} else if h.Stat.Finished() && oldFile.FileInfo.ModTime() != h.Stat.Fileinfo.ModTime() {
+	} else if h.State.Finished() {
```

Comment:
it's pretty difficult to follow logic of this method. Why remove the ModTime checks?